main.sh

```diff
@@ -29,18 +30,31 @@ if [[ "${GITHUB_REF:?}" != "refs/tags/"* ]]; then
 fi
 tag="${GITHUB_REF#refs/tags/}"
 
-# TODO: Support custom prefix of tags https://github.com/taiki-e/create-gh-release-action/issues/1
-if [[ ! "${tag}" =~ ^v?[0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z_0-9\.-]+)?(\+[a-zA-Z_0-9\.-]+)?$ ]]; then
+if [[ ! "${tag}" =~ ^${prefix}-?v?[0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z_0-9\.-]+)?(\+[a-zA-Z_0-9\.-]+)?$ ]]; then
     error "invalid tag format: '${tag}'"
     exit 1
 fi
-if [[ "${tag}" =~ ^v?[0-9\.]+-[a-zA-Z_0-9\.-]+(\+[a-zA-Z_0-9\.-]+)?$ ]]; then
+if [[ "${tag}" =~ ^${prefix}-?v?[0-9\.]+-[a-zA-Z_0-9\.-]+(\+[a-zA-Z_0-9\.-]+)?$ ]]; then
     prerelease="--prerelease"
 fi
-version="${tag#v}"
+
+version="${tag}"
+# extract the portion of the tag matching the prefix pattern
+if [[ ! "${prefix}" = "" ]]; then
+    prefix=$(expr match "${tag}" "\(${prefix}\)")
```

Comment:
shellcheck doesn't seem to like `expr` command.

```suggestion
    prefix=$(grep <<<"${tag}" -Eo "^${prefix}")
```

https://github.com/taiki-e/create-gh-release-action/runs/4582043847?check_suite_focus=true

> ```
> In main.sh line 44:
>     prefix=$(expr match "${tag}" "\(${prefix}\)")
>              ^--^ SC2003: expr is antiquated. Consider rewriting this using $((..)), ${} or [[ ]].
> 
> For more information:
>   https://www.shellcheck.net/wiki/SC2003 -- expr is antiquated. Consider rewr...
> ```

Reply:
ah, whoops!

i think we can also just change it to use `:` rather than `match`, which shellcheck [makes an exception for](https://github.com/koalaman/shellcheck/wiki/SC2003#exceptions): 

```suggestion
    prefix=$(expr "${tag}" : "\(${prefix}\)")
```

but...we can also get away with using grep here, as i'm sure the github actions runners will have grep installed.